src/Compilers/CSharp/Portable/Compilation/MemberSemanticModel.cs

```diff
@@ -1067,7 +1073,7 @@ private ImmutableArray<BoundNode> GuardedAddBoundTreeAndGetBoundNodeFromMap(CSha
 
             if (bound != null)
             {
-                alreadyInTree = _guardedNodeMap.ContainsKey(bound.Syntax);
+                alreadyInTree = _guardedNodeMap.ContainsKey(AsNodeMapKey(bound.Syntax));
```

Comment:
@mavasani it is not possible to handle this entirely at the mapping of bound nodes to IOperations, because the MemberSemanticModel currently always returns null if asked to bind any field initializer code block. This is a bug that needs fixing. @AlekseyTs has suggested (offline) introducing a bound node to represent the field initialization explicitly, and I'll try that.